core/src/main/java/org/infinispan/commands/write/InvalidateVersionsCommand.java

```diff
@@ -0,0 +1,150 @@
+package org.infinispan.commands.write;
+
+import org.infinispan.commands.VisitableCommand;
+import org.infinispan.commands.Visitor;
+import org.infinispan.container.DataContainer;
+import org.infinispan.container.versioning.InequalVersionComparisonResult;
+import org.infinispan.container.versioning.SimpleClusteredVersion;
+import org.infinispan.context.InvocationContext;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+/**
+ * Must be {@link VisitableCommand} as we want to catch it in persistence handling etc.
```

Comment:
@rvansa I only remember that I also suggested making it non-visitable, but I don't remember how you convinced me it's better to make it visitable :)